src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,14 @@
+package duke;
+
+public class DukeException extends Exception {
+    private final String error;
+
+    public DukeException(String error) {
+        this.error = error;
+    }
+
+    @Override
+    public String toString() {
+        return this.error;
+    }
+}
```

Comment:
you can try going further by having other exceptions extending to duke exceptions, so that you wont be always be throwing only duke exceptions. maybe DukeStorageSaveException for exceptions faced while saving and DukeStorageLoadException for exceptions faced while loading data and files.